File: Solutions/Python/1313.py
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        n = len(nums)
        output = []
        for i in range(n//2):
            freq , val = nums[2*i], nums[2*i+1]
            output += [val]*freq
        return output

File: Solutions/Java/1313.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int len = 0;
        for (int i = 0; i < nums.length; i += 2) {
            len += nums[i];
        }
        int[] result = new int[len];
        int initial = 0;
        for (int i = 0; i < nums.length; i += 2) {
            for (int j = 0; j < nums[i]; j++, initial++) {
                result[initial] = nums[i + 1];
            }
        }
        return result; 
    }
}
